protocol/backend/backend.go

```diff
@@ -150,9 +152,19 @@ func NewBackend(cfg *Config) (Backend, error) {
 		defaultTimeoutDuration = time.Hour
 	}
 
-	var addr ethcommon.Address
+	var (
+		addr   ethcommon.Address
+		txOpts *txsender.TxOpts
+		err    error
+	)
 	if cfg.EthereumPrivateKey != nil {
 		addr = common.EthereumPrivateKeyToAddress(cfg.EthereumPrivateKey)
+
+		// TODO: set gas limit and price?
```

Comment:
I was actually thinking that we should eliminate the `--gas-limit` flag from `swapd`.  The estimation works good, so I was thinking that it is just unnecessary complexity. In order to use a fixed gas limit across a wide array of transaction types requiring different amounts of gas, the user would have to set a gas limit much higher than what most individual transactions require. Are there other newish (non-legacy) programs with a similar gas limit option for multiple transaction types?

Reply:
the reason I originally added it was because certain networks (L2s primarily) don't work with geth's gas limit estimation, as geth always will estimate it too low, and the transactions always fail. I think we could probably put the gas limit in a per-network config, instead of having it as a CLI flag.

Reply:
Do those networks have a fork of the `go-ethereum` libraries that can do correct estimation?